core/src/wallet/common/synchronizers/AbstractBlockchainExplorerAccountSynchronizer.h

```diff
@@ -236,7 +237,16 @@ namespace ledger {
                 }).template flatMap<Unit>(account->getContext(), [buddy, self] (const Unit&) {
                     return self->synchronizeBatches(0, buddy);
                 }).template flatMap<Unit>(account->getContext(), [self, buddy] (const Unit&) {
-                    return self->_explorer->killSession(buddy->token.getValue());
+                    auto tryKillSession = Try<Future<Unit>>::from([=](){
+                        return self->_explorer->killSession(buddy->token.getValue());
+                    });
+                    if (tryKillSession.isFailure()) {
+                        buddy->logger->warn("Failed to delete synchronization token {} for account#{} of wallet {}",
+                                            static_cast<char *>(buddy->token.getValue()), buddy->account->getIndex(),
+                                            buddy->account->getWallet()->getName());
+                        return Future<Unit>::successful(unit);
```

Comment:
Why do you return a successful `Future<Unit>` when something bad occurs ? Is it any functional choice behind that stuff ?

Reply:
Yes, failing on sync token deletion should not be a "functional failure" that's why we return a successful `Unit` and raise a warning on that.
We are trying to figure out how to get rid of that sync token from Backend side e.g. include it in header or better have a solution from infra side.

Reply:
I suggest to add a comment when the code doesn't reflect the "normal intention" such as returning false positive case.